appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1554,11 +1556,17 @@ private void deployAll() throws GlassFishException {
                     contextRoot = null;
                 }
                 if (war.exists() && war.canRead()) {
+
                     if (war.getName().startsWith("ROOT.")) {
                         deployer.deploy(war, "--availabilityenabled=true", "--force=true", "--contextroot=/", "--loadOnly", "true");
                     } else {
-                        deployer.deploy(war, "--availabilityenabled=true", "--force=true", "--loadOnly", "true", "--contextroot", deployContext);
+                        if(deployContext == null) {
```

Comment:
deployContext is never null since it is set earlier from the war name.

Reply:
@smillidge regarding this, what's the purpose of the deployContext variable? Because it seems in these cases it would be more beneficial to just use contextRoot directly